src/components/login-page/LoginPage.jsx

```diff
@@ -0,0 +1,199 @@
+import CssBaseline from '@mui/material/CssBaseline';
+import Box from '@mui/material/Box';
+import { Button } from '@mui/material';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+import { useState } from 'react';
+// import { useNavigate } from 'react-router-dom';
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import Stack from '@mui/material/Stack';
+import { grey } from '@mui/material/colors';
+import { useAuth } from '../../utils/constants';
+import CircularProgress from '@mui/material/CircularProgress';
+import Snackbar from '@mui/material/Snackbar';
+
+export function LoginPage() {
+    // const navigate = useNavigate();
+
+    const inputsStyle = {
+        marginBottom: '21px',
+    };
+    const labelsStyle = {
+        marginBottom: '7px',
+    };
+    const mySecondaryColor = grey[500];
+
+    const [userInput, setUserInput] = useState({
+        username: '',
+        password: '',
+    });
+
+    const [isLoading, setIsLoading] = useState(false);
+    const [isDisabled, setIsDisabled] = useState(false);
+    const [alertMsg, setAlertMsg] = useState({
+        open: false,
+        vertical: 'top',
+        horizontal: 'center',
+    });
+    const { vertical, horizontal, open } = alertMsg;
+
+    const handleAlertOnClick = () => {
+        setAlertMsg({ ...alertMsg, open: true });
+    };
+
+    const handleAlertOnCLose = () => {
+        setAlertMsg({ ...alertMsg, open: false });
+    };
+
+    const auth = useAuth();
+    const handleLogingOnClick = () => {
+        setIsLoading(true);
+        setIsDisabled(true);

```

Comment:
if we want to reduce the number of states we have in this component, we can use the "isLoading" state to determine if a component should be disabled.

something like
\<my comp
  disabled={isLoading}
/>